### DISCUSSION TITLE
Simple PIXMTA example issue

### DISCUSSION THREAD

**Discussion by sherrick01821** [mdaus/nitro#397] [Q&A]
#include <iostream>
#include <nitf/TRE.hpp>

int main()
{

  nitf::TRE tre( "PIXMTA" , "PIXMTA" );

  tre.setField("NUMAIS", 1);

#if 1
  nitf::TREFieldIterator it = tre.begin();

  while( it != tre.end() )
{
  nitf::Pair pr = *it;

//  std::cout << pr.first() << "\n";
  ++it;
}
#endif

  tre.setField("AISDLVL[0]", 1);
  tre.setField("SCALE_X", "1.2");

}

When I compile and run the above example. It runs without error. However If I change the #if 1 to #if 0 an exception is thrown due to AISDLVL[0]. Why is that? Its a valid field

**Comment by sherrick01821** [comment]
As an added info.  The valid fields printed when using the std::cout when iterating over the TRE 

NUMAIS
AISDLVL[0]
ORIGIN_X
ORIGIN_Y
SCALE_X
SCALE_Y
SAMPLE_MODE
NUMMETRICS
PERBAND
RESERVED_LEN

**Comment by bradh** [comment]
I don't see the exception.
```
#include <nitf/TRE.hpp>

int main()
{
    nitf::TRE tre("PIXMTA", "PIXMTA");
    tre.setField("NUMAIS", 1);
    tre.setField("AISDLVL[0]", 1);
    tre.setField("SCALE_X", "1.2");
}
```
(which is essentially your code with the `#ifdef`'d part removed)

compiles and runs.

**Comment by sherrick01821** [comment]
That is odd.  After running cmake and then building the library with make and installing in /opt there was an empty /opt/include/std directory. Trying to compile this testpixmta.cpp example it complained about the nonexistence of <std/bit>  I then created a subdirectory ubder /opt/include called nitrostd and copied the files in externals/coda-oss/modules/c++/std/include/std to this directory and then the code compiled fine. There appears to be files in std created to satisfy C++ 11 compilers for C++ 17 like bit,  cstddef, filesystem  etc...
I also had to ignore the compiler warnings it complained about like in ImageSubheader.cpp  

 mutable nitf_Error error{};

where the compiler complained about no initializer.

What OS and compiler are you running?

**Reply by bradh** [reply]
I just did it in tree (as a unittest) and did a rebuild. I had to set the plugin path (environment variable), but that was all.
If you have problems with out of tree, maybe worth checking if the existing apps / tests work in the same environment.

Nothing special about the machine - Ubuntu 20.10 (aka "Groovy"), g++ 10.3.0.
